File: C#/PBTKata/Bank/Amount.cs
using LanguageExt;

namespace PBTKata.Bank
{
    public record Amount
    {
        public decimal Value { get; private init; }
        public static Option<Amount> From(decimal amount) =>
            amount > 0 ? new Amount { Value = amount } : Option<Amount>.None;

        public static bool operator >=(Amount a, Amount b) => a.Value >= b.Value;
        public static bool operator <=(Amount a, Amount b) => a.Value <= b.Value;
        public static bool operator >(Amount a, Amount b) => a.Value > b.Value;
        public static bool operator <(Amount a, Amount b) => a.Value < b.Value;
        public static Amount operator -(Amount a, Amount b) => a with { Value = a.Value - b.Value };
        public static Amount operator +(Amount a, Amount b) => a with { Value = a.Value + b.Value };
    }

    public static class AmountExtensions
    {
        public static Amount ToAmountUnsafe(this decimal amount) => Amount.From(amount).GetUnsafe();
    }
}

File: java/src/main/java/bank/Amount.java
package bank;

import io.vavr.control.Option;

import static io.vavr.control.Option.none;
import static io.vavr.control.Option.some;

public record Amount(double value) {
    public Amount {
        if (value < 0) throw new IllegalArgumentException("value must be a positive double");
    }

    public static Option<Amount> from(double amount) {
        return amount > 0 ? some(new Amount(amount)) : none();
    }
}